exporter/containerimage/annotations.go

```diff
@@ -0,0 +1,157 @@
+package containerimage
+
+import (
+	"fmt"
+
+	ocispecs "github.com/opencontainers/image-spec/specs-go/v1"
+
+	"github.com/containerd/containerd/platforms"
+	"github.com/moby/buildkit/exporter/containerimage/exptypes"
+)
+
+type Annotations struct {
+	Index              map[string]string
+	IndexDescriptor    map[string]string
+	Manifest           map[string]string
+	ManifestDescriptor map[string]string
+}
+
+type AnnotationsGroup map[string]*Annotations
+
+func ParseAnnotations(data map[string]string) (AnnotationsGroup, map[string]string, error) {
+	ag := make(AnnotationsGroup)
+	rest := make(map[string]string)
+
+	for k, v := range data {
+		if !exptypes.ExporterIsAnnotationKey(k) {
+			rest[k] = v
+			continue
+		}
+		if err := ag.set(k, v); err != nil {
+			return nil, nil, err
+		}
+	}
+	return ag, rest, nil
+}
+
+func ParseAnnotationsBytes(data map[string][]byte) (AnnotationsGroup, map[string][]byte, error) {
```

Comment:
Rather than doing the implementation twice the caller should just call `Parse(toBytesMap(m))`